src/iff.imageio/iffinput.cpp

```diff
@@ -385,48 +436,60 @@ IffInput::read_header()
+                                                || xmax >= m_header.width
+                                                || ymax >= m_header.height)
                                                 return false;
 
                                             // set tile width and height
-                                            m_iff_header.tile_width
-                                                = xmax - xmin + 1;
-                                            m_iff_header.tile_height
-                                                = ymax - ymin + 1;
+                                            m_header.tile_width = xmax - xmin
+                                                                  + 1;
+                                            m_header.tile_height = ymax - ymin
+                                                                   + 1;
 
                                             // done, return
                                             return true;
+                                        } else {
+                                            // skip to the next block.
+                                            if (!ioseek(chunksize, SEEK_CUR)) {
+                                                errorfmt(
+                                                    "IFF error io seek failed");
+                                                return false;
+                                            }
                                         }
-
-                                        // skip to the next block.
-                                        if (!ioseek(chunksize, SEEK_CUR))
-                                            return false;
                                     }
                                 } else {
                                     // skip to the next block.
-                                    if (!ioseek(chunksize, SEEK_CUR))
+                                    if (!ioseek(chunksize, SEEK_CUR)) {
+                                        errorfmt("IFF error io seek failed");
                                         return false;
+                                    }
                                 }
                             } else {
                                 // skip to the next block.
-                                if (!ioseek(chunksize, SEEK_CUR))
+                                if (!ioseek(chunksize, SEEK_CUR)) {
+                                    errorfmt("IFF error io seek failed");
                                     return false;
+                                }
                             }
                         }
                         // TBHD done, break
                         break;
+                    } else {
+                        // skip to the next block.
+                        if (!ioseek(chunksize, SEEK_CUR)) {
+                            errorfmt("IFF error io seek failed");
```

Comment:
The ImageInput helper methods ioseek() and ioread() handle calling errorfmt for you. If they fail, you just need to return false, you don't need to clutter with another error message call.